Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja

```diff
@@ -374,33 +358,46 @@ void {{ ClassName }}::Signal{{ UpperFirst(Property.attrib['Name']) }}({{ ', '.jo
 {%     set paramTypes   = [] %}
 {%     set paramDefines = [] %}
 {{     AutoComponentMacros.ParseRpcParams(Property, paramNames, paramTypes, paramDefines) }}
-            ->Method("{{ UpperFirst(Property.attrib['Name']) }}", [](const {{ ClassName }}* self, {{ ', '.join(paramDefines) }}) {
-                        self->m_controller->{{ UpperFirst(Property.attrib['Name']) }}({{ ', '.join(paramNames) }});
+            ->Method("{{ UpperFirst(Property.attrib['Name']) }}", []({{ ClassName }}* self, {{ ', '.join(paramDefines) }}) {
+{%                    if (InvokeFrom == 'Server')  %}
+                        self->{{ UpperFirst(Property.attrib['Name']) }}({{ ', '.join(paramNames) }});
+{%                    elif (InvokeFrom == 'Authority') or (InvokeFrom == 'Autonomous') %}
+                        if (self->m_controller)
+                        {
+                            self->m_controller->{{ UpperFirst(Property.attrib['Name']) }}({{ ', '.join(paramNames) }});
+                        }
+                        else
+                        {
+                            AZ_Warning("Network RPC", false, "{{ ClassName }} {{ UpperFirst(Property.attrib['Name']) }} method failed. Entity '%s' (id: %s) {{ ClassName }} is missing the network controller. This remote-procedure can only be invoked from {{InvokeFrom}} network entities, because this entity doesn't have a controller, it must not be a {{InvokeFrom}} entity. Please check your network context before attempting to call {{ UpperFirst(Property.attrib['Name']) }}.", self->GetEntity()->GetName().c_str(), self->GetEntityId().ToString().c_str())
+                        }
+{%                    endif %}
```

Comment:
@kberg-amzn does this honor the contract for Server as expected? Auth/Autonomous seem reasonable to me but I'm not as familiar with Server.

Reply:
Server->Auth is designed for cross-entity communication. EntityA wants to send EntityB a message.  EntityB might be an authority, or it might be a proxy on the server, EntityA doesn't know which, but only knows it needs to make sure EntityB's authority gets a message.

Reply:
Yup, Gene has the sum total of it.. server->auth is for all server cross entity communication, and these RPC's are implicitly 'trusted' as they are coming from another server that has presumably already done all the necessary security checks.